src/controllers/userController.js

```diff
@@ -0,0 +1,79 @@
+// Signup page controller
+import code from '../helpers/statusCode';
+import models from '../database/models';
+import findUser from '../services/findUser';
+import generateToken from '../helpers/generateToken';
+import Password from '../helpers/generatePassword';
+import Response from '../helpers/sendResponse';
+// import user from '../database/models/user';
+/** Class representing a User controller . */
+export default class usercontroller {
+  /**
+     * @description user signUp method
+     * @param {object} req
+     * @param {object} res
+     * @returns {object} createUser
+     * @memberof userController
+     */
+  static async signup(req, res) {
+    const {
+      firstName, lastName, email, password, confirmPassword
+    } = req.body;
+
+    const rawData = req.body;
+    if (req.body.password !== req.body.confirmPassword) {
+      return Response.error(res, code.badRequest, 'password mis-Match');
+    }
+    if (!firstName || !lastName || !email || !password || !confirmPassword) {
+      return Response.error(res, code.badRequest, 'please fill all fields');
+    }
+    // check if user exists
+    const userExist = await findUser.isUserExist(req.body.email);
+    if (userExist) {
+      return Response.error(res, code.conflict, 'User already exist');
+    }
+    // create the user
+    try {
+      const obj = new Password(rawData);
```

Comment:
Please pass the argument which is only needed by this function which is password.